Include all .xml files in targeting pack

wtgodbe opened the pull request:
Resolves https://github.com/dotnet/aspnetcore/issues/26068 and the 3.1 part of https://github.com/dotnet/aspnetcore/issues/26073. We were intentionally including the .xml files for project references & Extensions references, but not other package references. Also turns on the build of the targeting pack for next month's 3.1 servicing event.

wtgodbe committed:
Include all .xml files in targeting pack

wtgodbe committed:
Undo change to .js

wtgodbe committed:
Undo change to .js

wtgodbe commented:
CI won't build the ref pack, this test build will: https://dev.azure.com/dnceng/internal/_build/results?buildId=824004&view=results

wtgodbe committed:
Better syntax

Pilchie commented:
:eyes:

wtgodbe commented:
@Pilchie think it's worth trying to sneak this into 3.1.9? The build is currently at corefx. That said, this bug has existed since at least 3.0.0, so it's not necessarily urgent that we fix it today.

Pilchie commented:
Agree that it's not urgent.  I'm fine with including it if it doesn't introduce any risk, but it's not worth pushing hard for.

dougbu commented on Directory.Build.props:
```diff
@@ -84,10 +84,10 @@
     <IsTargetingPackBuilding Condition=" '$(DotNetBuildFromSource)' == 'true' ">false</IsTargetingPackBuilding>
     <IsTargetingPackBuilding
-        Condition=" '$(IsTargetingPackBuilding)' == '' AND '$(AspNetCorePatchVersion)' != '8' ">false</IsTargetingPackBuilding>
+        Condition=" '$(IsTargetingPackBuilding)' == '' AND '$(AspNetCorePatchVersion)' != '10' ">false</IsTargetingPackBuilding>
```
Insert swear words here

wtgodbe replied:
😭

dougbu replied:
I messed up the top-level discussion. Instead, _update_ changes to this file and Versions.props to target 3.1.9.

dougbu commented on src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:
```diff
@@ -147,9 +147,13 @@ This package is an internal implementation of the .NET Core SDK and is not meant
 
       <RefPackContent Include="@(AspNetCoreReferenceAssemblyPath)" PackagePath="$(RefAssemblyPackagePath)" />
-      <RefPackContent Include="@(AspNetCoreReferenceDocXml)" PackagePath="$(RefAssemblyPackagePath)" />
+      <RefPackContent Include="@(AspNetCoreReferenceDocXml->Distinct())" PackagePath="$(RefAssemblyPackagePath)" />
```
I'm less concerned about duplicate XML doc file paths than duplicate files in the final layout i.e. different doc files depending on which is copied second. Suggest copying `%(Filename)` from this item group into a temporary one and adding an `<Error/>` if its `->Count()` doesn't match `->Distinct()->Count()`.

wtgodbe replied:
There will be an error anyways if there are duplicates, which will have a more descriptive error message than our custom error, since it lists the actual duplicate (similar to the .pdb issue).

dougbu commented:
> CI won't build the ref pack

Huh❔ CI would have built the Microsoft.AspNetCore.App.Ref package if you had not changed how `$(IsTargetingPackBuilding)` is set. Revert that and this PR will build fine for 3.1.9.

wtgodbe commented:
> Huh❔ CI would have built the Microsoft.AspNetCore.App.Ref package if you had not changed how $(IsTargetingPackBuilding) is set. Revert that and this PR will build fine for 3.1.9.

I targeted 3.1.10 because the branch is closed for 3.1.9 changes. We'd need to get tactics approval to merge this late, which doesn't seem worth it.

dougbu commented:
> We'd need to get tactics approval to merge this late, which doesn't seem worth it.

:ok:

dougbu commented:
What about your internal build will cause Microsoft.AspNetCore.App.Ref to be built❔

wtgodbe commented:
> What about your internal build will cause Microsoft.AspNetCore.App.Ref to be built❔

I changed the `3.1.10` conditions to `3.1.9`

dougbu commented:
In that case, the regular validation build will do the needful 😃

wtgodbe commented:
> In that case, the regular validation build will do the needful 😃

The branch for the PR is versioned 3.1.9, but only builds the targeting pack when `Patch` is 10. In the internal build, I pushed another commit that changes the condition for the targeting pack so it builds when `Patch` is 9.

dougbu commented:
Lemme know when you've checked the artifacts from a build that should do the **Right Thing:tm:**

dougbu reviewed (approved):
Looks good though I'm curious why we need the `->Distinct()` you added&hellip;

wtgodbe committed:
Remove distinct

wtgodbe commented:
> Looks good though I'm curious why we need the ->Distinct() you added…

I added it defensively, but the behavior is the same without it. Just removed it (same in the other PR).

wtgodbe commented:
> Lemme know when you've checked the artifacts from a build that should do the Right Thing™️

Ref pack from the test build look good

dougbu commented:
Alright. This is ready for 3.1.10 but needs to sit 'til the branches open for that.
